xbmc/video/dialogs/GUIDialogVideoManagerExtras.cpp

```diff
@@ -100,47 +99,102 @@ void CGUIDialogVideoManagerExtras::AddVideoExtra()
           sources, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(),
           g_localizeStrings.Get(40015), path))
   {
-    std::string typeVideoVersion;
-    std::string videoTitle;
-    int idFile{-1};
-    int idMedia{-1};
-    MediaType itemMediaType;
-    VideoAssetType videoAssetType{VideoAssetType::UNKNOWN};
-
-    const int idVideoVersion{m_database.GetVideoVersionInfo(path, idFile, typeVideoVersion, idMedia,
-                                                            itemMediaType, videoAssetType)};
-
     const int dbId{m_videoAsset->GetVideoInfoTag()->m_iDbId};
     const VideoDbContentType itemType = m_videoAsset->GetVideoContentType();
 
-    if (idVideoVersion != -1)
+    const VideoAssetInfo newAsset{m_database.GetVideoVersionInfo(path)};
+
+    if (newAsset.m_idFile != -1 && newAsset.m_assetTypeId != -1)
     {
-      CFileItemList versions;
-      m_database.GetVideoVersions(itemType, dbId, versions, videoAssetType);
-      if (std::any_of(versions.begin(), versions.end(),
-                      [idFile](const std::shared_ptr<CFileItem>& version)
-                      { return version->GetVideoInfoTag()->m_iDbId == idFile; }))
+      unsigned int msgid;
```

Comment:
We should initialize the variable. Currently it is guaranteed that it will not be used uninitialized, but who knows for the future.

Reply:
I don't have a sensible default to provide though. 0 is as bad as a random message number. It's only used in the two blocks that popup a dialog, would that be OK if I moved the declaration right next to the use instead?
Same thing in the manage versions dialog.

Reply:
> 0 is as bad as a random message number

Up to you. I really do not want to spend much time to discuss this. I have the habit to init everything that has no default ctor, just to document in the code that I have taken care and did not simply forget.